verifier/verifier.go

```diff
@@ -0,0 +1,248 @@
+// Package Verifier provides an implementation of notation.Verifier interface
+package verifier
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/notaryproject/notation-go"
+	"github.com/notaryproject/notation-go/dir"
+	"github.com/notaryproject/notation-go/internal/envelope"
+	"github.com/notaryproject/notation-go/internal/plugin"
+	"github.com/notaryproject/notation-go/internal/plugin/manager"
```

Comment:
Can you update the code to use the refactored `plugin` package?
See https://github.com/notaryproject/notation-go/tree/main/plugin

Reply:
Code updated. verifier package now incorporates the refactored plugin and plugin/proto packages. 
Since PR #199 by @JeyJeyGao introduces the refactored PluginMock in internal/mocks.go, tests in the verifier package are utilizing it directly. After PR #199 is merged, this PR should rebase and work smoothly with the PluginMock.